accountClient/accountClient.go

```diff
@@ -46,6 +46,35 @@ func (c *E3dbAccountClient) InternalGetClientAccount(ctx context.Context, client
 	return result, err
 }
 
+func (c *E3dbAccountClient) RegisterClient(ctx context.Context, params ClientRegistrationRequest) (*ClientRegistrationResponse, error) {
+	var result *ClientRegistrationResponse
+	path := c.Host + "/" + AccountServiceBasePath + "/e3db/clients/register"
+	request, err := e3dbClients.CreateRequest("POST", path, params)
+	if err != nil {
+		return result, e3dbClients.NewError(err.Error(), path, 0)
+	}
+	resp, err := e3dbClients.ReturnE3dbServiceCall(c.E3dbAuthClient, ctx, request, &result)
+	if err != nil {
+		return result, e3dbClients.NewError(err.Error(), path, 0)
+	}
+	backupClient := resp.Header.Get("X-Backup-Client")
+	result.RootClientID = backupClient
```

Comment:
Looks to me like we can get rid of `ReturnE3dbServiceCall` if we pass `c.E3dbAuthClient.AuthHTTPClient(ctx)` to `MakeRawServiceCall` and have register client return the backup client for a newly registered client as part of the response object(which is I change we should make regardless of making this easier to refactor, @lkwdwrd any context on why that value is in a header instead of the response body?)

Reply:
not really. I think the idea is this was response meta rather than response data, hence the header. The response could be handled one way, and the backup ID could could be fetched separately if desired by the SDKs. At this point I don't think we should change it, since this would break our current SDKs. Going forward, I think this will shift to the returned client info, which is part of the response body. Until that time, let's not make a breaking change to the API.

Reply:
Right, I'm not arguing for making a breaking change, I'm suggesting we make an additive change(return the backup client as part of the new client service response), let the sdk continue to use the header value, but don't write any new code(including wherever we are calling this method) that expects this value to be present in a header.

Reply:
I guess I worry about breaking things like the Java and other strongly typed SDKs by adding values to the return... The JSON parsing is not as fault-tolerant as in go. We are using a different JSON library in the SDK than in the Scala code, so maybe it works better, but I know in the scala parser, if the JSON doensn't fit into the box perfectly it starts throwing errors...

@efabens - do you happen to know off the top of your head if the JSON parsing in the Java SDK would properly deal with an extra field in the JSON return in a registration response? (if not off the top of your head, don't worry about it).

(it will probably be OK. We did add an Access Key ID to record returns and nothing got too angry)

Reply:
For this PR I will do a no-op, and make a note to refactor once account service is updated